src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RequiredAttributes.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#if !NETCOREAPP
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Field | AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+    internal sealed class RequiredMemberAttribute : Attribute
```

Comment:
Are these used anywhere yet? Or is this for future use?

Reply:
These are required by the compiler to use `required` members. If we were targetting .NET 7, we wouldn't need this, but since we're not, they're the same as us defining `IsExternalInit` or all of the nullable analysis attributes.